### PR TITLE
LPD-27321

### PR DESCRIPTION
**Pull request by icgsbr**
# Jira Link
- [LPD-27321](https://liferay.atlassian.net/browse/LPD-27321)

### CODE REVIEW

File: modules/test/playwright/tests/object-web/objectActions.spec.ts [hunk 1]
```diff
@@ -71,17 +85,185 @@ test.describe('manage object actions through object actions tab', () => {
...
+	// Create object definition with an attachment field
+
+	const objectDefinition = await apiHelpers.objectAdmin.postObjectDefinition({
+		active: true,
+		externalReferenceCode: 'customObjectERC',
+		label: {
+			en_US: 'customobject',
+		},
+		name: 'CustomObject',
+		objectFields: [
```
Comment (cgmonte) [inline modules/test/playwright/tests/object-web/objectActions.spec.ts:115]: Hey! I think that this is too much information to be inside the test file. If every test that needs a specific definition define the definition inside the test file, then we will have a huge file that is hard to read. 

Please, create a .ts file inside the dependencies folder and declare and export an object definition inside that file where every key is an object field, like:
```
export const mockedObjectFields = {
       attchmentField = 	{
		DBType: 'Long',
		businessType: 'Attachment',
		externalReferenceCode: 'testAttachment',
		indexed: true,
		indexedAsKeyword: false,
		indexedLanguageId: '',
		label: {
			en_US: 'testAttachment',
		},
		listTypeDefinitionId: 0,
		name: 'testAttachment',
		objectFieldSettings: [
			{
				name: 'acceptedFileExtensions',
				value: 'jpeg, jpg, pdf, png, txt',
			},
			{
				name: 'maximumFileSize',
				value: 100,
			},
			{
				name: 'fileSource',
				value: 'userComputer',
			},
			{
				name: 'showFilesInDocumentsAndMedia',
				value: false,
			},
		],
		required: false,
		state: false,
		system: false,
		type: 'Long',
		unique: false,
	},
}
```
 Then import to this file and assign to `objectFields`:

`objectFields: [mockedObjectFields.attchmentField]`

Comment (cgmonte) [reply modules/test/playwright/tests/object-web/objectActions.spec.ts:115]: Also, try to use `postRandomObjectDefinition()` for a even more concise object creation.

File: modules/test/playwright/tests/object-web/objectActions.spec.ts [hunk 2]
```diff
@@ -71,17 +85,185 @@ test.describe('manage object actions through object actions tab', () => {
...
+			'do-not-replay@liferay.com'
+		);
+
+	createdEntities.notificationQueueEntryIds =
+		notificationQueueEntries.items.map((item: any) => item.id);
+
+	expect(notificationQueueEntries.items.length).toBeTruthy();
+});
+
+test.afterEach(async ({apiHelpers}) => {
```
Comment (cgmonte) [inline modules/test/playwright/tests/object-web/objectActions.spec.ts:231]: `test.afterEach` should come at the top of the file, just before the first call to a real test. In this case, before the call to `test.describe`.

File: modules/test/playwright/tests/object-web/objectActions.spec.ts [hunk 3]
```diff
@@ -71,17 +85,185 @@ test.describe('manage object actions through object actions tab', () => {
...
+
+	const notificationQueueEntries =
+		await apiHelpers.notification.getNotificationQueueEntriesPage(
+			'do-not-replay@liferay.com'
+		);
+
+	createdEntities.notificationQueueEntryIds =
+		notificationQueueEntries.items.map((item: any) => item.id);
+
+	expect(notificationQueueEntries.items.length).toBeTruthy();
```
Comment (cgmonte) [inline modules/test/playwright/tests/object-web/objectActions.spec.ts:228]: This assertion is not good. What if there is like ANY notification from `do-not-replay@liferay.com` from another test that was not properly cleaned up? We would have this passing. We need to assert if our specific notification is in the queue. 

One wait that I found was this: 

`expect(notificationQueueEntries.items.some((item) => item.recipientsSummary === notificationTemplate.recipients[0].to.en_US)).toBeTruthy();
`

it does the job but it looks ugly! I am sure you are going to find a more elegant way! :heart:

**Review by cgmonte — CHANGES_REQUESTED** [review]
Hi, Igor! What a pleasure was to review this PR! I just made some small requests for change. Check them out!

File: modules/test/playwright/tests/object-web/objectActions.spec.ts (hunk 2 above)
Comment (icgsbr) [reply modules/test/playwright/tests/object-web/objectActions.spec.ts:231]: Done!

File: modules/test/playwright/tests/object-web/objectActions.spec.ts (hunk 1 above)
Comment (icgsbr) [reply modules/test/playwright/tests/object-web/objectActions.spec.ts:115]: Done!

**Comment by icgsbr** [comment]
ci:test:sf

**Comment by icgsbr** [comment]
Hi @cgmonte ! All alterations done, please take a look in order to see if everything is according to what you requested. Thanks! :smile:

**Comment by icgsbr** [comment]
ci:test:sf

**Comment by icgsbr** [comment]
ci:test:sf

**Review by cgmonte — CHANGES_REQUESTED** [review]
Hi, Igor! I asked some questions and requested some changes. Pleas, let me kown what you think.

File: modules/test/playwright/tests/object-web/dependencies/objectMockedFields.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+export const mockedObjectFields: any = {
+	attachmentFieldUserComputer: {
```
[the remark this answers is not in the document]
Comment (icgsbr) [reply modules/test/playwright/tests/object-web/dependencies/objectMockedFields.ts:6]: We have two type of attachments, one that the user will upload from its own computer and the other that comes from documents and media. The documents and media one I'm already doing in another ticket; therefore, I tried to differentiate with the purpose of being more understandable.

**Comment by icgsbr** [comment]
Hi @cgmonte ! All the alterations were done. Thanks!

**Comment by icgsbr** [comment]
ci:test:sf

**Review by cgmonte — APPROVED** [review]
LGTM!

**Comment by cgmonte** [comment]
ci:test:playwright

**Comment by icgsbr** [comment]
Hi @cgmonte !

From what I saw the errors aren't related to the commits, could you please double-check if I'm right? I may have let something slip.

Best regards.

**Comment by icgsbr** [comment]
FYI, I'll be resolving the branch conflicts after making sure the test errors aren't related to the alterations done in this Pull Request.

**Comment by icgsbr** [comment]
ci:test:sf

**Comment by icgsbr** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by icgsbr** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:forward:force

**Pull request closed without merging**